src/app/pages/system/general-settings/ntp-server/ntp-server-list/ntp-server-list.component.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  Component, OnInit, ChangeDetectionStrategy, ChangeDetectorRef,
+} from '@angular/core';
+import { MatTableDataSource } from '@angular/material/table';
+import { untilDestroyed, UntilDestroy } from '@ngneat/until-destroy';
+import { TranslateService } from '@ngx-translate/core';
+import { filter } from 'rxjs/operators';
+import { NtpServer } from 'app/interfaces/ntp-server.interface';
+import { EmptyConfig, EmptyType } from 'app/pages/common/entity/entity-empty/entity-empty.component';
+import { NtpServerFormComponent } from 'app/pages/system/general-settings/ntp-server/ntp-server-form/ntp-server-form.component';
+import { WebSocketService, DialogService } from 'app/services';
+import { IxSlideInService } from 'app/services/ix-slide-in.service';
+
+@UntilDestroy()
+@Component({
+  selector: 'app-ntp-server-list',
+  templateUrl: './ntp-server-list.component.html',
+  styleUrls: ['./ntp-server-list.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class NtpServerListComponent implements OnInit {
+  dataSource: MatTableDataSource<NtpServer> = new MatTableDataSource([]);
+  displayedColumns = [
+    'address',
+    'burst',
+    'iburst',
+    'prefer',
+    'minpoll',
+    'maxpoll',
+    'actions',
+  ];
+  loading = true;
+  loadingConf: EmptyConfig = {
+    type: EmptyType.Loading,
+    large: false,
+    title: this.translate.instant('Loading...'),
+  };
+  emptyConf: EmptyConfig = {
+    type: EmptyType.NoPageData,
+    large: true,
+    title: this.translate.instant('No servers have been added yet'),
+  };
+  errorConf: EmptyConfig = {
+    type: EmptyType.Errors,
+    large: true,
+    title: this.translate.instant('Can not retrieve response'),
+  };
+  error = false;
+
+  get currentEmptyConf(): EmptyConfig {
+    if (this.loading) {
+      return this.loadingConf;
+    }
+    if (this.error) {
+      return this.errorConf;
+    }
+    return this.emptyConf;
+  }
+
+  constructor(
+    private ws: WebSocketService,
+    private cdr: ChangeDetectorRef,
+    private slideInService: IxSlideInService,
+    private dialog: DialogService,
+    private translate: TranslateService,
+  ) { }
+
+  ngOnInit(): void {
+    this.getData();
+  }
+
+  createDataSource(servers: NtpServer[]): void {
+    this.dataSource = new MatTableDataSource(servers);
+    this.cdr.markForCheck();
+  }
+
+  getData(): void {
+    this.ws.call('system.ntpserver.query').pipe(
+      untilDestroyed(this),
+    ).subscribe((servers) => {
+      this.loading = false;
```

Comment:
Don't see you setting `loading` to `true` anywhere.

Reply:
it is true by default for first data loading and then the table refreshes data without a spinner

Reply:
Would it work for subsequent calls, like when row is removed?

Reply:
yeah, `getData` will update the table without showing a spinner

Reply:
And why is this a good thing? :)

Reply:
Initially, I thought showing the loading spinner on each call of getData could be annoying, e.g., `slideIn.onClose$`. Updated `getData` method to set loading to true before each call. It seems `*matNoDataRow` is visible only when a data source is empty, so the loading spinner won't appear on adding/editing/deleting actions when rows exist.